services/file_create.proto

```diff
@@ -30,75 +37,102 @@ import "basic_types.proto";
+ * realms and multiple shards.
  *
- * The current API ignores shardID, realmID, and newRealmAdminKey, and creates everything in shard 0
- * and realm 0, with a null key. Future versions of the API will support multiple realms and
- * multiple shards.
+ * ### Block Stream Effects
+ * After the file is created, the FileID for it SHALL be returned in the
+ * transaction receipt, and SHALL be recorded in the transaction record.
  */
 message FileCreateTransactionBody {
     /**
-     * The time at which this file should expire (unless FileUpdateTransaction is used before then
-     * to extend its life)
+     * An expiration timestamp.
+     * <p>
+     * When the network consensus time exceeds this value, the network
+     * SHALL expire the file.
      */
     Timestamp expirationTime = 2;
 
     /**
-     * All keys at the top level of a key list must sign to create or modify the file. Any one of
-     * the keys at the top level key list can sign to delete the file.
+     * A list of keys that represent file "owners".
+     * <p>
+     * Every `Key` in this list MUST sign this `fileCreate` transaction, as
+     * well as any `fileUpdate` or `fileAppend` that modifies this file.<br/>
+     * At least one `Key` in this list MUST sign any `fileDelete` transaction
+     * to delete this file.<br/>
+     * If this `KeyList` is empty, the file SHALL be created immutable and the
+     * only field that may be changed subsequently is the `expirationTime`.
+     * An immutable file cannot be deleted except with a `systemDelete`
+     * transaction, or by expiration.
      */
     KeyList keys = 3;
 
     /**
-     * The bytes that are the contents of the file
+     * A byte array of file content.
+     * <p>
+     * The file SHALL be created with initial content equal to this field.
      */
     bytes contents = 4;
 
     /**
-     * Shard in which this file is created
+     * A shard in which this file is created
      */
     ShardID shardID = 5;
 
     /**
-     * The Realm in which to the file is created (leave this null to create a new realm)
+     * A realm in which this file is created.
+     * <p>
+     * The shard number for this realm MUST match the value in `shardID`.<br/>
+     * Currently, this MUST be `0` for both fields.<br/>
+     * If the desired realm is `0.0`, this SHOULD NOT be set.
      */
     RealmID realmID = 6;
 
     /**
-     * If realmID is null, then this the admin key for the new realm that will be created
+     * The "create realm" was never enabled, and
+     * should not be possible on file creation.<br/>
+     * An admin key for a new realm, if one is created.
+     * Added deprecated tag 2024-05.
      */
-    Key newRealmAdminKey = 7;
+    Key newRealmAdminKey = 7 [deprecated = true];
 
     /**
-     * The memo associated with the file (UTF-8 encoding max 100 bytes)
+     * A short description of this file.
+     * <p>
+     * This value, if set, MUST NOT exceed 100 bytes when encoded as UTF-8.
```

Comment:
Should we call out configuration properties when applicable? This value must no exceed transaction.maxMemoUtf8Bytes which has a default value of 100 bytes.

Reply:
Updated.